types/history/history-tests.ts

```diff
@@ -96,14 +119,14 @@ let input = { value: "" };
 }
 
 {
-    let location1 = LocationUtils.createLocation('path/1', 1);
```

Comment:
Trade off: you can no longer use non-object types

Reply:
does this mean that these types don't match the reality of the library?

Reply:
Correct, that's the trade off.

Ideally we would be use the same technique without forcing this to be an object type, but declaration merging doesn't allow that. For example, declaration merging works with interfaces:

```ts
interface Bar {}

interface Foo {
  bar: Bar;
}

interface Bar {
  a: number;
}

declare const foo: Foo;

foo.bar.a; // number
```

… but it will not override a type alias:

```ts
type Bar = unknown;

interface Foo {
  bar: Bar;
}

// Compile error
// Duplicate identifier 'Bar'
type Bar = number;

declare const foo: Foo;

foo.bar; // unknown
```

This does mean it's a breaking change, but it shouldn't be too difficult to workaround this by wrapping primitives in an object, like I did here (`2` -> `{ value: 2 }`).

Reply:
https://github.com/microsoft/TypeScript/issues/4062#issuecomment-360528573

Reply:
well, I'm not the maintainer so I'm not sure if I get a vote around here but I would vote against a change that means the type signatures no longer match with a valid input of the library

if it's critical to change the location state to only accept an object, it seems like that first needs to occur in the `history` library itself